python/ray/dag/compiled_dag_node.py

```diff
@@ -907,11 +900,37 @@ def run(self):
         monitor.start()
         return monitor
 
+    def _execute_until(
+        self,
+        execution_index: int,
+    ) -> Any:
+        """Repeatedly execute this DAG until the given execution index.
+
+        Args:
+            execution_index: The execution index to execute until.
+
+        Returns:
+            The execution result corresponding to the given execution index.
+        """
+        while execution_index >= len(self._result_buffer):
+            if self._buffered_results > MAX_BUFFER_COUNT:
+                raise ValueError(
+                    "Too many buffered results: call ray.get() on previous "
```

Comment:
Print the max buffered results here.

Also, this code is not sufficient to catch the case where the user is still holding onto refs, right? Can we detect that case? If not, it's okay to do as a follow-up, but let's please make a note of it.

Reply:
Updated exception message. Added TODO.